client.go

```diff
@@ -121,3 +121,18 @@ func remote(cmd string) error {
 
 	return nil
 }
+
+func store(key string, reader io.Reader) {
+	c, err := net.Dial(gSocketProt, gSocketPath)
+	if err != nil {
+		log.Printf("dialing to send server: %s", err)
+		return
+	}
+
+	fmt.Fprintf(c, "store %d %s\n", gClientID, key)
+	io.Copy(c, reader)
+	c.Close()
+
+	// wait for server to finish storing data and send response, if any
+	io.ReadAll(c)
```

Comment:
If the connection is slow then yes `lf` would hang until it's finished or there's some kind of error. The server doesn't really do anything much apart from handling connections and forwarding `send` commands, so from a performance point of view it is sufficiently fast and I don't notice any kind of delay unless I deliberately add in `time.Sleep` for testing purposes.

Storing the data in the same thread and waiting for the server to finish was my intention, otherwise there could be a race condition as you have described. Currently `app.runShell` continues anyway even storing the data fails (e.g. if I deliberately kill the server), do you think it would be better to return instead?

Reply:
I'm unsure what's best. For the data we are storing now, I'm OK with race conditions as I said. If we, theoretically, stored something like the selection this way, I'd want `app.runShell` to abort if anything is off.

I've never noticed performance problems with `lf -remote` commands, so maybe that's OK. I do run into https://github.com/gokcehan/lf/issues/495; I don't understand that issue thoroughly, but I think you'd know whether it would be a problem here.

Reply:
I'm generally allowing race conditions - although `maps` and `cmds` rarely change, the jump list will naturally accumulate entries. I have added another commit c55607a which checks if the `store` command succeeds, otherwise `app.runShell` is aborted, although in my testing I haven't encountered this scenario without deliberately killing the server.

#495 is a separate issue which only involves running `lf -remote` during startup - I posted a more detailed explanation here: https://github.com/gokcehan/lf/issues/1293#issuecomment-1586098749. That being said you can trigger a similar error if you add `maps` as a startup command, but I can't see why anyone would want to do this.